extjsdk/README.md

```diff
@@ -0,0 +1,93 @@
+
+#### <a name="closeHandler" id="closeHandler">Closure
+The closure handler does not deal with a specific message or type of message, but instead is called when either your code calls `client.close()` or the WebSocket connection is forced to close, most likely due to a problem with the connection. This handler is called after everything but the handlers are reset, essentially creating a new Client targeting the same source. If you have saved a reference to the Client's Listener, be aware that the Listener is stopped and replaced with a functionally identical Listener just before this handler is called. The closure handler receives the Client whose connection closed.
+
+#### Configuration
+The configuration handler receives the message containing a configuration document created on the Vantiq client. This message is only received upon a successful source connection, and contains the source's configuration document as a Map at `((Map) <message>.getObject()).get("config")`. The configuration handler receives an ExtensionServiceMessage.
+
+#### Reconnection
+Under certain circumstances, typically a change to the source's configuration document, the Vantiq server may disconnect from the source. The reconnection handler is called when this occurs. It is suggested that in this handler you reset any changes caused by the configuration document and then reconnect using `client.connectToSource()`. If you do not want to send the reconnect message yourself or do not wish to create your own handler, then calling `client.setAutoReconnect(true)` will tell the Client to automatically try to reconnect once any handlers are finished The reconnection handler receives an ExtensionServiceMessage.
+
+#### Publish
+The Publish handler is called when a message is received that was generated by Vantiq with `PUBLISH <object> TO SOURCE <source name>`. The published object can be retrieved through `<message>.getObject()`. Note that anything declared with the `USING` keyword will also be placed into the same location, e.g. `PUBLISH {"hello":"world"} TO SOURCE <sourceName> USING {"option":"one"}` will generate an object that looks like `{"hello":"world","option":"one"}`. The Publish handler receives an ExtensionServiceMessage.
+
+#### <a name="queryHandler" id="queryHandler"></a>Query
+The Query handler is called when a message is received that was generated by Vantiq with `SELECT <keys> FROM SOURCE <source name>`. Query messages expect a response, either [data](#queryResponse) or an [error](#queryError), so every Client has a default handler is created that sends back an error stating that no Query handler was set. Options specified using the `WITH` keyword are received as a Map obtained with `<message>.getObject()`.
```

Comment:
Might be worth noting that part of the reason for the default query handler is that not every extension source will necessarily support the `query` operation.  Thus, if you're source doesn't support it, leave the handler at its default setting.